### PR TITLE
fixes #2336 : added an option to directly add a jid to block list

### LINKED ISSUE #2336: When viewing the block list, there should be an option to add a JID
Currently to add a JID to the block list I have to open a conversation with that JID, then block them, then close the conversation.

When viewing the block list from account details it would be nice if there was a way to add a JID to the list directly without opening a conversation.

### PR DESCRIPTION
**Pull request by vikiCoder**
Added an option to directly add a jid to block list from BlocklistActivity

### CODE REVIEW

**Comment by iNPUTmice** [comment]
Hi,

Thank you for your PR.
is there a reason you made the Menu Item hideable via intent?

**Comment by vikiCoder** [comment]
Yes. Because the same menu is being used by other activity,  ChooseContactActivity.java, and this particular menu item is not desired to be shown in that other activity. (I hope I understood your question correctly)

**Comment by iNPUTmice** [comment]
But the BlocklistActivity is not used somewhere else, is it? So the blocklistActivity could always make that item visible, or am I missing something?

File: src/main/java/eu/siacs/conversations/ui/EditAccountActivity.java [hunk 1]
```diff
@@ -572,7 +572,7 @@ public boolean onCreateOptionsMenu(final Menu menu) {
 			if (!mAccount.getXmppConnection().getFeatures().blocking()) {
 				showBlocklist.setVisible(false);
 			} else {
-				showBlocklist.setEnabled(mAccount.getBlocklist().size() > 0);
+				showBlocklist.setEnabled(mAccount.getBlocklist().size() >= 0);
```
Comment (iNPUTmice) [inline src/main/java/eu/siacs/conversations/ui/EditAccountActivity.java]: This line can probably be omitted entirly. Testing for `>=0` will always return true.

Comment (vikiCoder) [inline src/main/java/eu/siacs/conversations/ui/EditAccountActivity.java]: yes that line can be omitted.

**Comment by vikiCoder** [comment]
And about visibility of that item, it can be done that instead of making it visible using intent, directly do it in BlocklistActivity. I'm updating both.

**Comment by iNPUTmice** [comment]
I don't think that dialog needs another confirmation dialog. Just block the Jabber ID directly.

If you change the menu item description to *Block Jabber ID* you can reuse the same string in the Enter JID Dialog (change it from Block Contact)

File: src/main/java/eu/siacs/conversations/ui/BlocklistActivity.java [hunk 2]
```diff
@@ -58,6 +66,43 @@ protected void filterContacts(final String needle) {
 		getListItemAdapter().notifyDataSetChanged();
 	}
 
+	@Override
+	public boolean onCreateOptionsMenu(final Menu menu) {
+		super.onCreateOptionsMenu(menu);
+		final Intent i = getIntent();
```
Comment (iNPUTmice) [inline src/main/java/eu/siacs/conversations/ui/BlocklistActivity.java]: this line doesn't do anything

Comment (vikiCoder) [inline src/main/java/eu/siacs/conversations/ui/BlocklistActivity.java]: Opps, sorry. Forgot to remove that. Its done now.

**Pull request merged**